File: Selenium/Project/CRMProject_Java/Activity8.java
package CRMProject_Java;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;

public class Activity8 {
    /*
    8. Traversing tables
        Goal: Open the accounts page and print the contents of the table.
            a. Open the browser to the Alchemy CRM site and login.
            b. Navigate to the Sales -> Accounts page.
            c. Find the table on the page and print the names of the first 5 odd-numbered rows
            of the table to the console.
            d. Close the browser.
     */
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setUp() {
        WebDriverManager.firefoxdriver().setup();

        //Instance of the driver
        driver = new FirefoxDriver();

        //Opening URL in browser
        driver.get("http://alchemy.hguy.co/crm");

    }

    @Test
    public void LoginToWebsite() {
        System.out.println("Activity3, Title Page of is : " + driver.getTitle());

        //login logic
        WebElement userName = driver.findElement(By.id("user_name"));
        WebElement passWord = driver.findElement(By.id("username_password"));

        //passing credentials
        userName.sendKeys("admin");
        passWord.sendKeys("pa$$w0rd");

        //find login button and click on it
        driver.findElement(By.id("bigbutton")).click();

        //New Page
        String AfterLoginPage = driver.findElement(By.id("tab0")).getText();
        System.out.println("After login, you should get dashboard : " + AfterLoginPage);

        //Assertions
        Assert.assertEquals(AfterLoginPage, "SUITECRM DASHBOARD");

    }

    @Test
    public void navigateToAccountsTable() throws InterruptedException {
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.findElement(By.id("grouptab_0")).click();

        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.findElement(By.id("moduleTab_9_Accounts")).click();
        String validSalesWindow = driver.findElement(By.className("recent_h3")).getText();
        //Assertions
        Assert.assertEquals(validSalesWindow, "Recently Viewed");
        System.out.println("New Window Opened, Confirmation text is :" + validSalesWindow);
        /*
        Thread.sleep(Long.parseLong("2000"));

        System.out.println("Waited for the event !!");

        driver.findElement(By.xpath("//span[@class='suitepicon suitepicon-action-info']")).click();
        Thread.sleep(Long.parseLong("2000"));
//        String pageSourceTxt=driver.getPageSource();
//        System.out.println("PageSource :" + pageSourceTxt);

        String phoneNumber = driver.findElement(By.xpath("//span[@class='phone']")).getText();
////        Alert iHelpButton = driver.switchTo().alert();
////        String iHelpButtontext = iHelpButton.getText();
        System.out.println("Prompt window, Phone Number is : " + phoneNumber);

         */

        //Think Time
        Thread.sleep(2000);

        //look for table
        String h2Text = driver.findElement(By.className("module-title-text")).getText();
        System.out.println("header 2 is : " + h2Text);

        //WebElement rowsdata = driver.findElement(By.xpath("//table[@class='list view table-responsive']/tbody/tr[contains(@class, 'oddListRowS1')]/td[contains(@class, 'inlineEdit')]"));
//        for (int i = 0; i <5 ; i++){
//            WebElement rowsdata = driver.findElement(By.xpath("/html/body/div[5]/div/div[3]/form[2]/div[3]/table/tbody/tr[2]")
//            System.out.println(rowsdata.getText());
//        }

        for(int i=1;i<=13;i=i+2){

            WebElement rowdata = driver.findElement(By.xpath("//table/tbody/tr["+i+"]/td[3]/b"));
            System.out.println(rowdata.getText());
        }

    }
    @AfterClass
    public void tearDown(){
        driver.quit();
    }
}

File: Selenium/Project/CRM_SeleniumProject_Python/Activity8.py
from selenium import webdriver
import time
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("http://alchemy.hguy.co/crm")

    # Print the title of the page
    print("Page title is: ", driver.title)

    # Login Logic
    userName = driver.find_element(By.ID, "user_name")
    passWord = driver.find_element(By.ID, "username_password")

    userName.send_keys("admin")
    passWord.send_keys("pa$$w0rd")

    driver.find_element(By.ID, "bigbutton").click()

    # after login
    afterLoginPage = driver.find_element(By.ID, "tab0")
    print("Dashboard Name : ", afterLoginPage.text)

    time.sleep(15)

    # Navigate to accounts table

    driver.find_element(By.ID, "grouptab_0").click()
    time.sleep(10)

    driver.find_element(By.ID, "moduleTab_9_Accounts").click()
    time.sleep(10)

    for i in [1,3,5,7,9,11,13]:
        rowData = driver.find_element(By.XPATH, f"//table/tbody/tr[{i}]/td[3]/b")
        print(rowData.text)


    driver.quit()

